test/test.ts

```diff
@@ -1062,3 +1062,25 @@ test('should verify timeout overrides passed to add', async t => {
 
 	await queue.onIdle();
 });
+
+test('should skip an aborted job', t => {
+	const queue = new PQueue();
+
+	const controller = new AbortController();
+
+	controller.abort();
+	queue.add(() => {
```

Comment:
An aborted job will never ever resolve. Perhaps it should reject instead?

Reply:
What do other promise APIs that support AbortController do in this case?

Reply:
`window.fetch` throws `Uncaught (in promise) DOMException: The user aborted a request.`

Reply:
Yeah, should abort then. Just use a normal error for now, with a TODO comment to use https://nodejs.org/api/errors.html#abort_err